app/components/more-details.jsx

```diff
@@ -0,0 +1,125 @@
+// https://github.com/EnCiv/civil-pursuit/issues/89
+
+'use strict'
+import React, { useEffect, useState } from 'react'
+import { createUseStyles } from 'react-jss'
+import cx from 'classnames'
+import { JsonForms } from '@jsonforms/react'
+import { vanillaCells, vanillaRenderers } from '@jsonforms/vanilla-renderers'
+import { PrimaryButton } from './button.jsx'
+import { rankWith, isEnumControl } from '@jsonforms/core'
+import { withJsonFormsControlProps } from '@jsonforms/react'
+
+const CustomSelectRenderer = withJsonFormsControlProps(({ data, handleChange, path, uischema, schema }) => {
+  const classes = useStyles({ mode: 'light' })
+  const options = schema.enum || []
+  const label = uischema.label || schema.title
+
+  return (
+    <div>
+      <label>{label}</label>
+      <select value={data || ''} onChange={ev => handleChange(path, ev.target.value)} className={classes.select}>
+        <option value="" disabled>
+          Choose one
+        </option>
+        {options.map(option => (
+          <option key={option} value={option}>
+            {option}
+          </option>
+        ))}
+      </select>
+    </div>
+  )
+})
+
+const customRenderers = [...vanillaRenderers, { tester: rankWith(3, isEnumControl), renderer: CustomSelectRenderer }]
+
+const MoreDetails = props => {
+  const {
+    className = 'Submit Form',
+    schema = {},
+    uischema = {},
+    details = {},
+    onDone = () => {
+      valid, value
+    },
+    value,
+  } = props
+  const [data, setData] = useState(details)
+  const [isValid, setIsValid] = useState(false)
+  const classes = useStyles(props)
+
+  const handleOnClick = () => {
+    onDone({ valid: true, value })
+  }
+
+  const handleIsValid = () => {
+    const requiredData = ['householdIncome', 'housing', 'numberOfSiblings']
+    return requiredData.every(i => data[i])
+  }
+
+  useEffect(() => {
+    setIsValid(handleIsValid())
+  }, [data])
+
+  return (
+    <div className={cx(classes.moreDetailsContainer, className)}>
+      <p className={classes.title}>We just need a few more details about you to get started.</p>
+      <div className={classes.formContainer}>
+        <div>
+          <JsonForms
+            schema={schema}
+            uischema={uischema}
+            data={data}
+            renderers={customRenderers}
+            cells={vanillaCells}
+            onChange={({ data, _errors }) => setData(data)}
+          />
+        </div>
+        <PrimaryButton
+          primary
+          tile={'Submit form'}
+          className={classes.submitButton}
+          onClick={() => handleOnClick()}
+          onDone={onDone}
```

Comment:
this should be more like `onDone={()=>onDone({valid: isValid, value: data})}`
although submit won't get called unless isValid it's just an informative way to say it.  Also, we need to send the data the user filled out up to the next level.